### ISSUE TITLE
Cursor moves when backtracking

### ISSUE THREAD

**Issue by jesper-bengtson** [hannesm/Kopitiam#19]
Whenever the buffer backtracks, the cursor is moved to the end of the new green section. This is not optimal. A common way of working is to just start typing or deleting in the middle of the green section to fix something. The buffer will naturally backtrack to the closest point to the edit point as possible but, since the cursor is moved as well, text is added in the wrong place (or worse deleted). 

This is not that much of an issue when moving forward (it might even be desirable to have the cursor move but I would have to think about that some more).

So to sum up, when the buffer backtracks due to an edit, don't move the cursor and don't switch the focus. When stepping back and forwards using buttons or hot keys, then it's fine to leave it the way it is.
